anydex/wallet/node/node.py

```diff
@@ -0,0 +1,188 @@
+import json
+import logging
+import socket
+from enum import Enum, auto
+from ipaddress import ip_address, IPv4Address
+from time import time
+
+from anydex.config import get_anydex_configuration
+
+_logger = logging.getLogger(__name__)
+
+
+class Source(Enum):
+    """
+    Enum representing possible provider source for nodes.
+    Currently supports nodes provided by users or by AnyDex.
+    """
+
+    DEFAULT = auto()
+    USER = auto()
+
+
+class Cryptocurrency(Enum):
```

Comment:
I'm noticing that this class contains all cryptocurrencies that we set out to implement and that are already implemented (except Dash?). However, for the cryptocurrencies supported in bitcoinlib (BTC, LTC, DASH), a slightly different system is already in place, where the cryptocurrency is only represented by a `string`, rather than this class. Do you think it would be appropriate to change those implementations to support this structure as well? If not, should the aforementioned cryptocurrencies be omitted from here?

Reply:
If we plan on being able to choose node for all cryptocurrencies, including the bitcoinlib ones, is it easier and structure wise acceptable to implement node definition on our own for bitcoinlib cryptocurrencies, or can we adjust bitcoinlib implementations to this to avoid re-implementing node selection?

Found this for bitcoinlib
[https://bitcoinlib.readthedocs.io/en/latest/_static/manuals.setup-bitcoind-connection.html](url)

Reply:
I've moved the Cryptocurrency Enum out of the `node`-package into the parent package `wallet`. I've also added the missing cryptocurrency Dash. For consistency sake it is likely useful to create methods accepting types of this Enum rather than just strings. Also raw strings of the cryptocurrencies can then be replaced with Cryptocurrency.BITCOIN.value for example.